extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/ForwardedParser.java

```diff
@@ -167,6 +146,29 @@ private void calculate() {
             }
         }
 
+        String forwarded = delegate.getHeader(FORWARDED);
+        if (forwardingProxyOptions.allowForwarded && forwarded != null) {
+            String forwardedToUse = forwarded.split(",")[0];
```

Comment:
You can't split it here, if you check the RFC (https://datatracker.ietf.org/doc/rfc7239/) the following is a a valid `Forwarded` header: `Forwarded: for=192.0.2.43, for=198.51.100.17;by=203.0.113.60;proto=http;host=example.com"`

If we do this split, before the matcher is run, we will make the `forwardedToUse` have the wrong value, sine it will split and use just `for=192.0.2.43` and will disregard `host` and `proto`

Reply:
It's better to create an issue describing that, otherwise the PR loses its intention.

Reply:
@ejba I agree, lets have this PR focused on this specific issues of dealing with both types of forward headers.
@luneo7 Can you please open another issue related to this split as well as some other optimizations you've been suggesting in other comments, thanks

Reply:
I would rather fix this right away since they are all related, the code is already done and working, instead of opening more PRs and getting more reviews. This PR will touch the code either way. Breaking it, would just make it take longer to fix it all.

Reply:
This is a fix, not an improvement

Reply:
@ejba Sorry, if it is already done...  then I'm not 100% sure if we should revert. I guess it would be safer given the sensitivity. So if you are OK, then please follow with another minor PR related to this split as advised by @luneo7, it will minimize the risk of some side-effects, thanks